federation/cmd/federation-controller-manager/app/controllermanager.go

```diff
@@ -129,12 +156,54 @@ func Run(s *options.CMServer) error {
 		glog.Fatal(server.ListenAndServe())
 	}()
 
-	run := func() {
+        eventBroadcaster := record.NewBroadcaster()
+        eventBroadcaster.StartLogging(glog.Infof)
+        eventBroadcaster.StartRecordingToSink(&unversionedcore.EventSinkImpl{Interface: kubeClient.Core().Events("")})
+        recorder := eventBroadcaster.NewRecorder(api.EventSource{Component: "federation-controller-manager"})
+
+	//run := func() {
+        run := func(stop <-chan struct{}) {
 		err := StartControllers(s, restClientCfg)
 		glog.Fatalf("error running controllers: %v", err)
 		panic("unreachable")
 	}
-	run()
+
+        if !s.LeaderElection.LeaderElect {
+                run(nil)
+                panic("unreachable")
+        }
+
+        id, err := os.Hostname()
+        if err != nil {
+                return err
+        }
+
+        // TODO: enable other lock types
+        rl := resourcelock.EndpointsLock{
+                EndpointsMeta: api.ObjectMeta{
+                        Namespace: "kube-system",
+                        Name:      "kube-controller-manager",
```

Comment:
This is not the latest code. I had changed "Namespace" and "Name" to "federation" and "federation-controller-manager". For some reason the latest code didn't get pushed. I will push the latest shortly. But, @nikhiljindal the rest is the same as kube controller manager

@madhusudancs since we don't have Endpoints API at federation I used a k8s clientset to the k8s cluster hosting the federation pods